en/orm/retrieving-data-and-resultsets.rst

```diff
@@ -193,6 +193,21 @@ key.
 .. note::
 
     The ``first()`` method will return ``null`` if no results are found.
+    
+
+.. warning::
+
+    All instances returned from the same unmodified query is referenced to one another. Modify the query to create a new unreferenced instance.
```

Comment:
I'm hesitant to add a warning for this as it isn't a scenario that many people will encounter, and if we include every potential edge case in the documentation as a warning. The documentation would become noisy with all the potential edge cases.

Could we instead inline this information into paragraphs above that talk about fetching rows?

Reply:
based on my post in discourse, it seems that other users like myself were expecting a different & un-referenced instance when calling `$query->first()` again. The closest note I could find about this is [`Once executed, modifying and re-evaluating a query will result in additional SQL being run.`](https://book.cakephp.org/4/en/orm/query-builder.html#queries-are-lazily-evaluated), but doesn't explicitly state that results are referenced if query is unmodified.

I feel a warning is good to prevent un-expected results. Query is core and important.

or maybe it should be added @ [queries-are-lazily-evaluated ?](https://book.cakephp.org/4/en/orm/query-builder.html#queries-are-lazily-evaluated)

Reply:
> or maybe it should be added @ queries-are-lazily-evaluated ?

Would something like 'Calling first() multiple times will return same reference.' added to the bullet point on first() in that section help?